docs/developer/sound.md

```diff
@@ -0,0 +1,129 @@
+# Music and Sounds
+
+## Melodies
+
+The high level music API in arcade uses a syntax for creating melodies from
+sequences of notes. Melodies are authored using the `music.Melody` class
+in arcade which encodes them in a string syntax. For some example melodies,
+see the [mixer package](https://github.com/Microsoft/pxt-common-packages/blob/master/libs/mixer/melody.ts#L387).
+
+Melodies are constructed with a series of commands outlined below.
+
+### Command: play note
+
+To play a note:
+```
+[ABCDEFG] (\d+)  (:\d+)  (-\d+)
+note      octave length  tempo
+```
+
+The octave, length, and tempo are all optional parameters. Here are some examples:
+
+* `A5:2-120` (play note `A` in octave `5` for `2` beats and set the tempo to `120` bpm)
+* `C` (play note `C` for one beat at the current tempo)
+* `g5:1` (play note `G` in octave `5` for `1` beat at the current tempo)
+
+### Command: rest
+
+To rest:
+```
+[R] (:\d+)
+```
+For example:
+
+* `R` (rest for `1` beat at current tempo)
+* `R:3` (rest for `3` beats at current tempo)
+
+### Command: play tone
+
+To play a frequency for a duration in ms:
+```
+!\d+,\d+
+```
+
+For example:
+* `!262,500` (play frequency `262` Hz for `500` ms)
+
+### Command: set envelope
+
+To set the ADSR envelope for the played sounds:
+```
+@\d+,\d+,\d+,\d+
+```
+
+Where the numbers are:
```

Comment:
"Where the 4 values indicate:"

This `\d+` notation is confusing to me. Maybe this is something I don't know about ("decimal, > 0")?...Ah, wait, JS string formatting.